plugins/ag-grid-generate-example-files/src/executors/generate/executor.ts

```
@@ -92,6 +95,9 @@ async function getProvidedFiles(folderPath: string) {
 export async function generateFiles(options: ExecutorOptions, gridOptionsTypes: Record<string, GridOptionsType>) {
     const isDev = options.mode === 'dev';
     const folderPath = options.examplePath;
+    const folders = folderPath.split('/');
+    const pageName = folders[folders.length - 3];
+    const exampleName = folders[folders.length - 1];
```

Might be worth pulling this out to somewhere sensible, so if we change the structure in the future this doesn't suddenly break.

---

Putting the logic inside website docs (where the other `fileUtils` live), which is outside of the `plugins` folder, is not that straight forward because the example generation is dependant on the website docs, and would cause a circular dependency. It's also only grid specific.

Instead I've pulled it out as a function within plugins and throw an error if it's empty. Hopefully, it fails loud enough if the structure changes, so we know to change it in future